packages/cli/source/manifest/compiler.ts

```diff
@@ -1,33 +1,60 @@
-/* eslint-disable @typescript-eslint/no-unused-vars */
-/* eslint-disable @typescript-eslint/no-explicit-any */
+import { App, Manifest, Query, Resource } from "@hypertool/common";
 
 import yaml from "js-yaml";
 import fs from "fs";
 import path from "path";
+import joi from "joi";
 import chalk from "chalk";
+import { constants } from "@hypertool/common";
+import lodash from "lodash";
 
-import { paths, getMissingKeys, constants } from "../utils";
-import type { Manifest, App, Query, Resource } from "../types";
-
-const IDENTIFIER_REGEX = /^[a-zA-Z_][a-zA-Z_0-9-]+[a-zA-Z_0-9]$/;
+import { paths } from "../utils";
 
 let errorCount = 0;
 
-export const logMissingKeys = (
-    regsitryType: string,
-    keys: string[],
-    filePath = "<anonymous>",
-) => {
-    errorCount++;
-    console.log(
-        `${chalk.red(
-            "[error]",
-        )} ${filePath}: The following keys are missing in ${regsitryType}: ${keys.join(
-            ", ",
-        )}\n`,
-    );
-};
+const IDENTIFIER_REGEX = /^[a-zA-Z_][a-zA-Z_0-9-]+[a-zA-Z_0-9]$/;
 
+/**
+ * Joi Validation Schema for App, Query, Resource and Manifest
+ */
+const appSchema = joi.object({
+    name: joi.string().max(128).regex(IDENTIFIER_REGEX).required(),
+    title: joi.string().max(256).regex(IDENTIFIER_REGEX).required(),
+    slug: joi.string().max(128).regex(IDENTIFIER_REGEX).required(),
+    description: joi.string().max(512).allow("").default(""),
+    groups: joi
+        .array()
+        .items(joi.string().regex(IDENTIFIER_REGEX))
+        .default(["default"]),
+});
+
+const querySchema = joi.object({
+    name: joi.string().max(128).regex(IDENTIFIER_REGEX).required(),
+    description: joi.string().max(1024).allow("").default(""),
+    resource: joi.string().regex(IDENTIFIER_REGEX).required(),
+    content: joi.string().max(10240).required(),
+});
+
+const resourceSchema = joi.object({
+    name: joi.string().max(256).regex(IDENTIFIER_REGEX).required(),
+    description: joi.string().max(512).allow("").default(""),
+    type: joi
+        .string()
+        .valid(...constants.resourceTypes)
+        .required(),
+    connection: joi.any().required(),
```

Comment:
Is there a way to limit to string or object? Because any allows values like numbers, null, and so on.